massdriver.yaml

```diff
@@ -0,0 +1,172 @@
+schema: draft-07
+name: "azure-vpn-gateway"
+description: "Azure VPN Gateway is a service that uses a specific type of virtual network gateway to send encrypted traffic between an Azure virtual network and on-premises locations over the public Internet."
+source_url: github.com/massdriver-cloud/azure-vpn-gateway
+access: public
+type: infrastructure
+
+params:
+  required:
+    - gateway
+    - network
+  properties:
+    gateway:
+      type: object
+      title: Gateway Configuration
+      description: "**Provisioning of the VPN Gateway takes approximately 30-60 minutes.**"
+      required:
+        - generation
+        - auth_type
+      properties:
+        generation:
+          type: string
+          title: Gateway Generation
+          description: The generation for the VPN gateway. [Learn more](https://learn.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-about-vpn-gateway-settings#gateway-sku-by-performance).
+          enum:
+            - Generation1
+            - Generation2
+        auth_type:
+          type: string
+          title: Authentication Type
+          description: The authentication type to use for the VPN connection. **If using Microsoft Entra ID, be sure to authorize the Azure VPN app in your Azure account [here](https://login.microsoftonline.com/common/oauth2/authorize?client_id=41b23e61-6c1e-4545-b367-cd054e0ed4b4&response_type=code&redirect_uri=https://portal.azure.com&nonce=1234&prompt=admin_consent).** The authentication type **cannot be changed** after deployment.
+          $md.immutable: true
+          oneOf:
+            # - title: Certificate
```

Comment:
This will probably look weird in the UI. A dropdown w/ 1 field. I'm guessing this is where you plan on supporting certificate output as well?

Reply:
Looks like this:
![image](https://github.com/massdriver-cloud/azure-vpn-gateway/assets/8037512/5a2a968c-0c93-45bd-bd81-948ddaa00f30)